components/match2/commons/starcraft_starcraft2/match_group_util_starcraft.lua

```diff
@@ -327,30 +360,32 @@ function StarcraftMatchGroupUtil.constructSubmatch(games, match)
 	}
 end
 
--- Determine if a match has details that should be displayed via popup
+---Determine if a match has details that should be displayed via popup
+---@param match StarcraftMatchGroupUtilMatch
+---@return boolean
 function StarcraftMatchGroupUtil.matchHasDetails(match)
 	return match.dateIsExact
-		or match.vod
+		or String.isNotEmpty(match.vod)
 		or not Table.isEmpty(match.links)
-		or match.comment
-		or match.casters
+		or String.isNotEmpty(match.comment)
+		or String.isNotEmpty(match.casters)
 		or 0 < #match.vetoes
 		or Array.any(match.games, function(game)
 			return game.map and game.map ~= 'TBD'
-				or game.winner
+				or Logic.isNumeric(game.winner)
 		end)
 end
 
---[[
-Determines if any players in an opponent are not playing their main race by
-comparing them to a reference opponent. Returns the races played if at least
-one player chose an offrace or nil if otherwise.
-]]
+---Determines if any players in an opponent are not playing their main race by comparing them to a reference opponent.
+---Returns the races played if at least one player chose an offrace or nil if otherwise.
+---@param gameOpponent StarcraftMatchGroupUtilGameOpponent
+---@param referenceOpponent StarcraftStandardOpponent|StarcraftMatchGroupUtilGameOpponent
```

Comment:
Why the two inputs?

Reply:
if this function is called to determine offraces in a match it is a `StarcraftStandardOpponent`
if it is called the offraces in a submatch it is a `StarcraftMatchGroupUtilGameOpponent`